ls1intum/Apollon
# Add contributor guides for realtime collaboration

### Checklist
- ~~I documented the TypeScript code using JSDoc style~~ (NA).
- ~~I added multiple screenshots/screencasts of my UI changes~~ (NA)
- ~~I translated all the newly inserted strings into German and English~~ (NA)

### Motivation and Context

Currently, the realtime collaboration system is only explained in code, without an overview present in the contributor guidelines section of Apollon documentation, potentially confusing developers who might want to extend / modify the system. Additionally, the system imposes some constraints on the schema of Apollon diagrams, which affects development in other areas (for example, adding new diagram types) as well, while these constraints aren't documented in an accessible location.

### Description

This PR adds contributor guides for realtime collaboration system, providing a general overview of how the system is designed, alongside detailing constraints that the realtime collaboration system imposes on diagram schemas and guidelines for following them.

### Steps for Testing

1. Clone this branch
2. Build the docs (see the README)
3. The "Realtime Collaboration" section should exist in the documentation, accessible from the first page, listed under "Contributor Guide"
  - Not to be confused with the "Realtime Collaboration" section under "User Guide"
4. The newly added section should address the concerns described in here by providing a general overview of the realtime collaboration system in a clear and concise manner.
